quickwit/quickwit-serve/src/elasticsearch_api/rest_handler.rs

```diff
@@ -481,9 +487,22 @@ async fn es_compat_index_field_capabilities(
     Ok(search_response_rest)
 }
 
-fn convert_hit(hit: quickwit_proto::search::Hit, append_shard_doc: bool) -> ElasticHit {
-    let fields: BTreeMap<String, serde_json::Value> =
+fn convert_hit(
+    hit: quickwit_proto::search::Hit,
+    append_shard_doc: bool,
+    _source_excludes: &Option<Vec<String>>,
+    _source_includes: &Option<Vec<String>>,
+) -> ElasticHit {
+    let mut fields: BTreeMap<String, serde_json::Value> =
```

Comment:
I don't think this will work with a field like `app.id` where the JSON document is `{"app": {"id": 123}}`

Reply:
We could do it at the `fetch_docs_in_split` level, we won't have to traverse all the json doc by doing it on a `NamedFieldDocument`.

Reply:
NamedFieldDocument is not flattened, we would still need to traverse the doc

Reply:
updated to handle paths